File: Part 1 - Data Preprocessing/data_preprocessing_template_1.R
setwd("C:\\Users\\Adi\\Desktop\\Machine Learning A-Z Template Folder\\Part 1 - Data Preprocessing")

dataset <- read.csv("Data.csv")

# taking care of the missing data 

dataset$Age <- ifelse(test = is.na(dataset$Age), 
                      ave(x = dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)), dataset$Age)
dataset$Salary <- ifelse(test = is.na(dataset$Salary), 
                      ave(x = dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)), dataset$Salary)

# encoding categorical variables

dataset$Country <- factor(x = dataset$Country, levels = c("France", "Spain", "Germany"), labels = c(1,2,3))
dataset$Purchased <- factor(x = dataset$Purchased, levels = c("No", "Yes"), labels = c(0,1))

# divide data into train and test set 
library(caTools)
set.seed(123)
split=sample.split(dataset$Purchased, SplitRatio = 0.8)
training_set <- subset(dataset, split== TRUE)
test_set <- subset(dataset, split== FALSE)

# feature scaling
training_set[,2:3] = scale(training_set[,2:3])
test_set[,2:3] = scale(test_set[,2:3])

File: Part 1 - Data Preprocessing/data_preprocessing_template_1.py
# -*- coding: utf-8 -*-
"""
Created on Thu Mar 12 20:09:13 2020

@author: Adi
"""

# import library

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# import dataset
dataset= pd.read_csv('Data.csv')
x= dataset.iloc[:,:-1].values
y = dataset.iloc[:,3].values

#taking care of missing data
from sklearn.preprocessing import Imputer
imputer = Imputer(missing_values="NaN", strategy="mean",axis=0)
imputer = imputer.fit(x[:,1:3])
x[:,1:3] = imputer.transform(x[:,1:3])

# encoding categorical data
from sklearn.preprocessing import LabelEncoder, OneHotEncoder 
labelencoder_x= LabelEncoder()
x[:,0] = labelencoder_x.fit_transform(x[:,0])
onehotencoder= OneHotEncoder(categorical_features=[0])
x= onehotencoder.fit_transform(x).toarray()


labelencoder_y= LabelEncoder()
y = labelencoder_y.fit_transform(y)

# splitting the data into train and test set 
from sklearn.cross_validation import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size= 0.2, random_state= 0)

# Feature scaling 
from sklearn.preprocessing import StandardScaler
sc_X= StandardScaler()
x_train= sc_X.fit_transform(x_train)
x_test= sc_X.transform(x_test)